plugin/trino-opa/src/test/java/io/trino/plugin/opa/OpaAccessControlFilteringUnitTest.java

```diff
@@ -0,0 +1,362 @@
+                                        """.formatted(requestedColumnsForTable.getKey().getTableName(), column))))
+                .collect(toImmutableList());
+        assertStringRequestsEqual(expectedRequests, this.mockClient.getRequests(), "/input/action");
+        assertTrue(
+                Maps.difference(
+                        result,
+                        ImmutableMap.builder()
+                                .put(tableOne, ImmutableSet.of("table_one_column_one", "table_one_column_two"))
+                                .put(tableTwo, ImmutableSet.of("table_two_column_two"))
+                                .buildOrThrow()).areEqual());
+    }
+
+    @Test
+    public void testEmptyFilterColumns()
+    {
+        SchemaTableName someTable = SchemaTableName.schemaTableName("my_schema", "my_table");
+        Map<SchemaTableName, Set<String>> requestedColumns = ImmutableMap.of(someTable, ImmutableSet.of());
+
+        Map<SchemaTableName, Set<String>> result = authorizer.filterColumns(
+                requestingSecurityContext,
+                "my_catalog",
+                requestedColumns);
+
+        assertTrue(mockClient.getRequests().isEmpty());
+        assertTrue(result.isEmpty());
+    }
+
+    @Test
+    public void testFilterFunctions()
+    {
+        SchemaFunctionName functionOne = new SchemaFunctionName("my_schema", "function_one");
+        SchemaFunctionName functionTwo = new SchemaFunctionName("my_schema", "function_two");
+        Set<SchemaFunctionName> requestedFunctions = ImmutableSet.of(functionOne, functionTwo);
+        this.mockClient.setHandler(buildHandler("/input/action/resource/function/functionName", "function_two"));
+
+        Set<SchemaFunctionName> result = authorizer.filterFunctions(
+                requestingSecurityContext,
+                "my_catalog",
+                requestedFunctions);
+        assertEquals(ImmutableSet.copyOf(result), ImmutableSet.of(functionTwo));
+
+        List<String> expectedRequests = requestedFunctions.stream()
+                .map(function -> """
+                    {
+                        "operation": "FilterFunctions",
+                        "resource": {
+                            "function": {
+                                "catalogName": "my_catalog",
+                                "schemaName": "%s",
+                                "functionName": "%s"
+                            }
+                        }
+                    }""".formatted(function.getSchemaName(), function.getFunctionName()))
+                .collect(toImmutableList());
+        assertStringRequestsEqual(expectedRequests, this.mockClient.getRequests(), "/input/action");
+    }
+
+    @ParameterizedTest(name = "{index}: {0}")
+    @MethodSource("io.trino.plugin.opa.FilteringTestHelpers#emptyInputTestCases")
+    public void testEmptyRequests(
+            BiFunction<OpaAccessControl, SystemSecurityContext, Collection> callable)
+    {
```

Comment:
We have been removing Parameterized tests throughtout the code base because they are difficult to understand and debug because the test cases are separated from the test code, and they are often over abstracted into generic functions like this, so a reader can not know what is actually being tested.  Please just inline the cases.